text/3804-cfg-alias.md

```diff
@@ -0,0 +1,225 @@
+- Feature Name: `cfg_alias`
+- Start Date: 2025-04-23
+- RFC PR: [rust-lang/rfcs#3804](https://github.com/rust-lang/rfcs/pull/3804)
+- Rust Issue:
+  [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+
+[summary]: #summary
+
+This RFC introduces a way to name configuration predicates for easy reuse
+throughout a crate.
+
+```rust
+#![cfg_alias(x86_linux = all(
+    any(target_arch = "x86", target_arch = "x86_64"), target_os = "linux"
+))]
+
+#[cfg(x86_linux)]
+fn foo() { /* ... */ }
+
+#[cfg(not(x86_linux))]
+fn foo() { /* ... */ }
+```
+
+# Motivation
+
+[motivation]: #motivation
+
+It is very common that the same `#[cfg(...)]` options need to be repeated in
+multiple places. Often this is because a `cfg(...)` needs to be matched with a
+`cfg(not(...))`, or because code cannot easily be reorganized to group all code
+for a specific `cfg` into a module. The solution is usually to copy a `#[cfg]`
+group around, which is error-prone and noisy.
+
+Adding aliases to config predicates reduces the amount of code that needs to be
+duplicated, and giving it a name provides an easy way to show what a group of
+configuration is intended to represent.
+
+Something to this effect can be done using build scripts. This requires reading
+various Cargo environment variables and potentially doing string manipulation
+(for splitting target features), so it is often inconvenient enough to not be
+worth doing. Allowing aliases to be defined within the crate and with the same
+syntax as the `cfg` itself makes this much easier.
+
+Another benefit is the ability to easily adjust configuration to many different
+areas of code at once. A simple example is gating unfinished code that can be
+toggled together:
+
+```rust
+#![cfg_alias(todo = false)] // change `false` to `true` to enable WIP code
+
+#[cfg(todo)]
+fn to_be_tested() { /* ... */ }
+
+
+#[test]
+#[cfg(todo)]
+fn test_to_be_tested() { /* ... */ }
+```
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+There is a new crate-level attribute that takes a name and a `cfg` predicate:
+
+```rust
+#![cfg_alias(some_alias = predicate)]
+```
+
+`predicate` can be anything that usually works within `#[cfg(...)]`, including
+`all`, `any`, and `not`.
```

Comment:
I'd probably drop ", including..." here as it makes it more rather than less ambiguous.  Alternatively, it'd be OK to say e.g. "including (but not limited to) combining operators such as `all`, `any`, and `not`.